src/util/tab-events.js

```diff
@@ -1,57 +1,59 @@
 import eventToPromise from './event-to-promise'
 
+function tabChanged(tabId) {
+    return new Promise((resolve, reject) => {
+        browser.webNavigation.onCommitted.addListener((details)=>{
+            if(details.tabId == tabId && details.frameId == 0)
+                reject();
+        });
+        browser.webNavigation.onHistoryStateUpdated.addListener((details)=>{
+            if(details.tabId == tabId && details.frameId == 0)
+                reject();
+        });
+    });
+}
```

Comment:
My question about the choice of events: why do you choose to listen to onCommited and onHistoryStateUpdated? Is it ensured that these will capture all cases where a tab 'unloads' its page? There's a few more options in the [`webNavigation` API](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/webNavigation), and also listening to `tabs.onChange` to check for URL changes would be an option that comes to mind. Did you try or research any of these options?

I guess you may have chosen these because the activity logger also listens to them (when #50 is resolved). Listening to the same events could then indeed solve the problem of accidentally capturing the wrong page, so that's a smart solution. If it is possible it would be nice however to ensure we have a watertight solution, that could also be trusted to work for potential other scenarios.